rca/schools/models.py

```diff
@@ -237,44 +334,44 @@ def page_nav(self):
             {"title": "Contact"},
         ]
 
-    def format_page_teasers(self, obj):
-        if not obj:
+    def get_related_projects(self):
+        """
+        Displays latest projects related to this school page.
+        Returns:
+            List of:
+                filtered and formatted manually related ProjectPages
+                or automatically fetch project pages from project > school relationship
+        """
+        from rca.projects.models import ProjectPage
+
+        manual_related_projects = [
+            i.page.id for i in self.manually_related_project_pages.all()
+        ]
+        auto_related_projects = ProjectPage.objects.filter(
+            related_school_pages__page_id=self.id
+        )
+        if manual_related_projects:
+            return format_projects_for_gallery(
+                ProjectPage.objects.filter(id__in=manual_related_projects)
+            )
+        elif auto_related_projects:
+            return format_projects_for_gallery(auto_related_projects)
+
+    def get_student_research(self, student_research, request):
+        if not student_research:
             return
-        page_teasers = {"title": obj.title, "summary": obj.summary, "pages": []}
-        for item in obj.pages:
-            for block in item.value:
-                if block.block_type == "custom_teaser":
-                    page_teasers["pages"].append(
-                        {
-                            "title": block.value["title"],
-                            "description": block.value["text"],
-                            "image": block.value["image"],
-                            "link": block.value["link"]["url"],
-                            "type": block.value["meta"],
-                        }
-                    )
-                elif block.block_type == "page":
-                    page = block.value.specific
-                    summary = (
-                        page.introduction
-                        if hasattr(page, "introduction")
-                        else page.listing_summary
-                    )
-                    image = (
-                        page.hero_image
-                        if hasattr(page, "hero_image")
-                        else page.listing_image
-                    )
-                    page_teasers["pages"].append(
-                        {
-                            "title": page.title,
-                            "description": summary,
-                            "image": image,
-                            "link": page.url,
-                        }
-                    )
+        if student_research.link_page:
+            link = student_research.link_page.get_url(request)
+        elif student_research.link_url:
+            link = student_research.link_url
 
-        return page_teasers
+        link = student_research.link_page or student_research.link_url
```

Comment:
Isn't this repeating the previous `if` statement?

Reply:
:+1: fixed here https://github.com/torchbox/rca-wagtail-2019/pull/510/files#diff-6642bbb1b7771cb3e228f87b1a3d0e995ef2455634dd14f17d8076890956a875R365